funcx_endpoint/funcx_endpoint/endpoint/interchange.py

```diff
@@ -321,52 +321,85 @@ def _main_loop(self):
-                    log.debug("alive")
-                    last = time.time()
-
-                self.results_ack_handler.check_ack_counts()
-
-                try:
-                    incoming_task = self.pending_task_queue.get(
-                        block=True, timeout=0.01
-                    )
-                    task = try_convert_from_messagepack(incoming_task)
-                    log.debug(f"Submitting task : {task}")
-                    executor.submit_raw(task)
-                except queue.Empty:
-                    pass
-                except Exception:
-                    log.exception("Unhandled issue while waiting for pending tasks")
-
-                try:
-                    results = self.results_passthrough.get(False, 0.01)
-
-                    task_id = results["task_id"]
-                    if task_id:
-                        self.results_ack_handler.put(task_id, results["message"])
-                        log.info(f"Passing result to forwarder for task {task_id}")
-
-                    message = try_convert_to_messagepack(results["message"])
-
-                    # results will be a packed EPStatusReport or a packed Result
-                    log.debug(f"Publishing message {message}")
-                    results_publisher.publish(message)
-                except queue.Empty:
-                    pass
-
-                except Exception:
-                    log.exception(
-                        "Something broke while forwarding results from executor "
-                        "to forwarder queues"
-                    )
-                    continue
+
+            num_tasks_forwarded = 0
+            num_results_forwarded = 0
+
+            def process_pending_tasks():
+                nonlocal num_tasks_forwarded
+                while not self._quiesce_event.is_set():
+                    try:
+                        incoming_task = self.pending_task_queue.get(timeout=1)
+                        task = try_convert_from_messagepack(incoming_task)
+                        executor.submit_raw(task)
+                        num_tasks_forwarded += 1  # Safe given GIL
+
+                    except queue.Empty:
+                        continue
+
+                    except Exception:
+                        log.exception("Unhandled issue while waiting for pending tasks")
+                log.debug("Exit process-pending-tasks thread.")
+
+            def process_pending_results():
+                nonlocal num_results_forwarded
+                while not self._quiesce_event.is_set():
+                    try:
+                        results = self.results_passthrough.get(timeout=1)
+
+                        task_id = results["task_id"]
+                        if task_id:
+                            self.results_ack_handler.put(task_id, results["message"])
+                            log.debug(f"Passing result to forwarder for task {task_id}")
```

Comment:
While we're in this part of the code, can we remove the reference to "forwarder" and refer to the result processor instead (or maybe just a blanket reference to "cloud services" or something similar)? Anticipating the conversation we have next time we onboard someone and they see logs referring to a forwarder and ask what that means